developer/ruby/GitHubIssueStats.rb

```diff
@@ -0,0 +1,148 @@
+require 'github_api'
+require 'date'
+require 'yaml'
+
+# Get the begin date, by finding the last major/minor (X.Y.0) release
+# that is at least a week old
+def get_begin_date_and_previous_tag()
+  a_week_ago = Time.now - (60*60*24*7)
+
+  @github.repos.releases.list(owner: @repo_owner,
+                              repo: @repo).each_page do |page|
+    page.each do |release|
+      next if release.tag_name !~ /^v\d\.\d\.0$/
+      release_date = Time.parse(release.created_at)
+      next if release_date > a_week_ago
+      puts "Found previous major/minor release: #{release.tag_name}, #{release_date}"
+      return release_date, " (#{release.tag_name})"
+    end
+  end
+  puts "Cannot find previous release, setting time to 2005"
+  return Time.new(2005, 01, 01), ""
+end
+
+def get_num(issue)
+  issue.html_url.split('/')[-1].to_i
+end
+
+def get_issue_num(issue)
+  "\##{get_num(issue)}"
+end
+
+def get_html_url(issue)
+  issue.html_url
+end
+
+def get_title(issue)
+  issue.title
+end
+
+def print_issue(issue)
+  is_feature = false
+  issue.labels.each {|label| is_feature = true if label.name == "Enhancement Request"}
+
+  if is_feature
+    "- ![Improved:][improved] [#{get_issue_num(issue)}]( #{get_html_url(issue)} ), #{get_title(issue)}"
+  else
+    "- ![Fixed:][fixed] [#{get_issue_num(issue)}]( #{get_html_url(issue)} ), #{get_title(issue)}"
+  end
+end
```

Comment:
Not following what `![Improved:][improved]` is supposed to be. It seems halfways between an Image `![legend](link)` and a icon or something? @macumber

Reply:
These were buttons on a page a long time ago, I guess when we hosted release notes outside of GitHub.  There is no need to keep them